src/geom/circle.rs

```diff
@@ -9,37 +9,25 @@ use std::{
 #[derive(Clone, Copy, Default, Debug, Deserialize, Serialize)]
 ///A circle with a center and a radius
 pub struct Circle {
-    /// The x coordinate of the center
-    pub x: f32,
-    /// The y coordinate of the center
-    pub y: f32,
+    /// The position of the center of the circle
+    pub pos: Vector,
     /// The radius of the circle
     pub radius: f32,
 }
 
 impl Circle {
-    /// Create a new circle with the given dimensions
-    pub fn new<T: Scalar>(x: T, y: T, radius: T) -> Circle {
-        Circle {
-            x: x.float(),
-            y: y.float(),
-            radius: radius.float(),
-        }
-    }
-
     /// Create a circle with the center as a vector
-    pub fn newv<T: Scalar>(center: Vector, radius: T) -> Circle {
+    pub fn new<V: Into<Vector>>(center: V, radius: impl Scalar) -> Circle {
```

Comment:
So I really hate to ask you to change this because I think it's great, but I think this would read better if `center` was `impl Into Into<Vector>` rather than a separate generic.

Reply:
Ah, I thought that that was not possible. Sure, I will do that